src/software/thunderscope/binary_context_managers.py

```diff
@@ -58,6 +59,13 @@ def __init__(
         self.friendly_colour_yellow = friendly_colour_yellow
         self.full_system_proc = None
 
+        try:
+            self.thread = Thread(target=__restart__)
+            self.thread.start()
+            self.thread.join()
+        except Exception:
+            raise Exception("Did not properly restart FullSystem.")
```

Comment:
you don't need the try, except. Just start the create the thread here (but don't call `thread.start()`)